packages/sync/src/createClient.ts

```diff
@@ -30,6 +30,7 @@ export const createClient = async (userConfig?: DataSyncConfig): Promise<Voyager
     storage,
     clientConfig.mutationsQueueName);
   await offlineMutationHandler.replayOfflineMutations();
+  offlineQueueLink.openQueueOnNetworkStateUpdates();
```

Comment:
[Idea] Add setTimeout with 3/5 seconds delay?
If not maybe we can await?

Reply:
Is it to wait for queue being opened and all operations sent, so that user won't send operation while queue is being processed? This could happen also later whenever is queue opened.  To prevent this, can we enqueue all incoming operations while queue is being processed?

Reply:
https://github.com/aerogear/aerogear-js-sdk/blob/master/packages/sync/src/links/OfflineQueueLink.ts#L110

Tells me that you getting notification when things are enqueued.

Reply:
so it will be something like:
```javascript
new Promise(res,rej){
  mutate(..)
  listener(()=>{res()})
```
}